content/api/commands/then.md

```diff
@@ -51,20 +51,23 @@ Pass a function that takes the previously yielded subject as its first argument.
 
 ### Yields [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Subject-Management)
 
-`.then()` is modeled identically to the way Promises work in JavaScript.
 Whatever is returned from the callback function becomes the new subject and will
-flow into the next command (with the exception of `undefined`).
-
-Additionally, the result of the last Cypress command in the callback function
-will be yielded as the new subject and flow into the next command if there is no
-`return`.
-
-When `undefined` is returned by the callback function, the subject will not be
-modified and will instead carry over to the next command.
-
-Just like Promises, you can return any compatible "thenable" (anything that has
-a `.then()` interface) and Cypress will wait for that to resolve before
-continuing forward through the chain of commands.
+flow into the next command (with the exception of `undefined` or `null`). If the
+return value is a Promise or other "thenable" (anything with a `.then()`
```

Comment:
```suggestion
return value is a Promise or a Chained `.then()` command
```
doesn't it way for any returned cypress command?

Reply:
That's not actually what the docs are trying to say - it means literally "if the return value is a promise or something like `return { then: () => {}}`."

With that said, I do think it is extremely unclear, and am updating the wording to be more straightforward.